From 16ad54eec032d73a9d15a870c2b01d135c19fbd6 Mon Sep 17 00:00:00 2001
From: "Qualcomm Innovation Center, Inc" <quic_linux@quicinc.com>
Date: Thu, 21 May 2026 03:29:12 -0700
Subject: [PATCH] QCLINUX: arm64: dts: qcom: lemans: Add SAIL mailbox nodes

Add the SAIL mailbox IPCC controller and sail-mailbox device nodes to
lemans.dtsi for testing the sail-mailbox OOT kernel module.

The memory reservations (sail_mailbox_mem, sail_ota_mem) are already
present in the upstream lemans.dtsi.

Also add IPCC_CLIENT_SAIL0/1/2/3 client ID definitions to the
qcom-ipcc dt-bindings header.

Signed-off-by: Qualcomm Innovation Center, Inc. <quic_linux@quicinc.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/qcom/lemans.dtsi    | 37 +++++++++++++++++++++++++
 include/dt-bindings/mailbox/qcom-ipcc.h |  4 +++
 2 files changed, 41 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/lemans.dtsi b/arch/arm64/boot/dts/qcom/lemans.dtsi
index 61a89ca..6adc616 100644
--- a/arch/arm64/boot/dts/qcom/lemans.dtsi
+++ b/arch/arm64/boot/dts/qcom/lemans.dtsi
@@ -1156,6 +1156,43 @@
 			#mbox-cells = <2>;
 		};
 
+		ipcc_computeL1: mailbox@488000 {
+			compatible = "qcom,ipcc";
+			reg = <0x0 0x00488000 0x0 0x1000>;
+			interrupts = <GIC_SPI 231 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-controller;
+			#interrupt-cells = <3>;
+			#mbox-cells = <2>;
+		};
+
+		sail_mailbox: sail-mailbox@1ffe02c {
+			compatible = "qcom,sail-mailbox";
+			reg = <0x0 0x01FFE02C 0x0 0x10>,
+			      <0x0 0x01FFD018 0x0 0x10>,
+			      <0x0 0x17C0000C 0x0 0x04>;
+			mboxes = <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x2>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x3>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x4>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x5>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x6>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL0 0x7>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL2 0x8>,
+				 <&ipcc_computeL1 IPCC_CLIENT_SAIL1 0x9>;
+			memory-region = <&sail_mailbox_mem>,
+					<&sail_ota_mem>;
+			interrupt-parent = <&ipcc_computeL1>;
+			interrupts = <IPCC_CLIENT_SAIL0 0x2 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL0 0x3 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL0 0x4 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL0 0x5 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL0 0x6 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL0 0x7 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL2 0x8 IRQ_TYPE_EDGE_RISING>,
+				     <IPCC_CLIENT_SAIL1 0x9 IRQ_TYPE_EDGE_RISING>;
+			sail-handshake-delay = <50000>;
+			status = "okay";
+		};
+
 		qfprom: efuse@784000 {
 			compatible = "qcom,sa8775p-qfprom", "qcom,qfprom";
 			reg = <0x0 0x00784000 0x0 0x3000>;
diff --git a/include/dt-bindings/mailbox/qcom-ipcc.h b/include/dt-bindings/mailbox/qcom-ipcc.h
index dc07ca2..abf45bd 100644
--- a/include/dt-bindings/mailbox/qcom-ipcc.h
+++ b/include/dt-bindings/mailbox/qcom-ipcc.h
@@ -33,6 +33,10 @@
 #define IPCC_CLIENT_NSP1		18
 #define IPCC_CLIENT_TME			23
 #define IPCC_CLIENT_WPSS		24
+#define IPCC_CLIENT_SAIL0		27
+#define IPCC_CLIENT_SAIL1		28
+#define IPCC_CLIENT_SAIL2		29
+#define IPCC_CLIENT_SAIL3		30
 #define IPCC_CLIENT_GPDSP0		31
 #define IPCC_CLIENT_GPDSP1		32
 
-- 
2.34.1

